catalog/IndexScheme.cpp

```diff
@@ -58,16 +55,12 @@ IndexScheme* IndexScheme::ReconstructFromProto(const serialization::IndexScheme
 
   for (int index_num = 0; index_num < proto.index_entries_size(); ++index_num) {
     const serialization::IndexScheme_IndexEntry &index_entry = proto.index_entries(index_num);
-    std::vector<IndexSubBlockDescription> index_descriptions;
-    for (int i = 0; i < index_entry.index_description_size(); ++i) {
-      index_descriptions.emplace_back(index_entry.index_description(i));
-    }
     // Make sure that index with same name does not already exist.
     DCHECK(index_scheme->index_map_.find(index_entry.index_name())
            == index_scheme->index_map_.end())
         << "Attempted to create IndexScheme from proto with duplicate index names.";
-    // Store the index_name and corresponding list of index descriptions in map.
-    index_scheme->index_map_.emplace(index_entry.index_name(), std::move(index_descriptions));
+    // Store the index_name and corresponding index description in map.
+    index_scheme->index_map_.emplace(index_entry.index_name(), std::move(index_entry.index_description()));
```

Comment:
I don't know why the compiler allows `move` on `index_entry`, a `const` referenced object, but I would remove `move`. It is ok to call copy constructor here.